src/app/public/modules/modal/modal-adapter.service.ts

```diff
@@ -11,24 +16,44 @@ export class SkyModalAdapterService {
 
   private docRef: any;
   private bodyEl: HTMLElement;
+  private renderer: Renderer2;
+  private hostDomElem: Element;
 
+  // NOTE: In future breaking change - remove optional from RendererFactory2 paramater.
+  // Currently there to avoid a breaking change
   constructor(
-    private windowRef: SkyWindowRefService) {
-      this.docRef = this.windowRef.getWindow().document;
-      this.bodyEl = this.windowRef.getWindow().document.body;
+    private windowRef: SkyWindowRefService,
+    @Optional() private rendererFactory?: RendererFactory2
+  ) {
+    // RendererFactory2 is optional to avoid breaking changes for consumer unit tests.
+    // However, due to this being core functionality we are throwing an error
+    if (!this.rendererFactory) {
+      throw ('Please inject RenderFactory2');
```

Comment:
These errors aren't that descriptive and may cause more confusion.

On second thought, let's just remove the error throwing altogether and address any unit test issues as they come up (I doubt anyone is mocking this service).

In the future, it's still a good idea to make new parameters optional, and do our best to make the methods work even if the parameter is not supplied. In this case, it is just too difficult to meet the second requirement.

Thanks @Blackbaud-TrevorBurch.